server/src/actors/outcome.rs

```diff
@@ -0,0 +1,312 @@
+//! This module contains the actor that forwards Outcomes to a kafka topic
+//!
+
+use std::sync::Arc;
+use std::time::Instant;
+
+use actix::prelude::*;
+use chrono::{DateTime, SecondsFormat, Utc};
+use serde::{Serialize, Serializer};
+
+use semaphore_common::Config;
+use semaphore_general::protocol::EventId;
+use semaphore_general::reason::OutcomeReason;
+
+use crate::utils::instant_to_system_time;
+use crate::ServerError;
+
+/// Message
+#[derive(Debug, Serialize)]
+pub struct KafkaOutcomeMessage {
+    /// The timespan of the event outcome.
+    #[serde(serialize_with = "instant_serializer")]
+    pub timestamp: Instant,
+    /// Organization id.
+    pub org_id: Option<u64>,
+    /// Project id.
+    pub project_id: Option<u64>,
+    /// The DSN project key id.
+    pub key_id: Option<u64>,
+    /// The outcome.
+    pub outcome: Outcome,
+    /// Reason for the outcome.
+    pub reason: OutcomeReason,
+    /// The event id.
+    pub event_id: Option<EventId>,
+}
+
+impl Message for KafkaOutcomeMessage {
```

Comment:
Can we decouple the interface of the outcomes actor from the internal serialization to JSON? That would allow us to create a cleaner interface for the outcomes enum, by merging the outcome and the reason into one enum.

Reply:
ok, done